Add signing rewritten assemblies

russcam opened the pull request:
This commit adds the ability to sign the rewritten assembly
when running on Full Framework. Signing is not currently
supported by Mono.Cecil on Net Core.

When a keyfile is supplied and merge is not specified (false),
the keyfile will be used to sign the rewritten assembly. When
merge is specified, the keyfile will be used to sign the merged
assembly.

Add net472 as a TargetFramework to allow signing functionality.

Move from Mono.Options to CommandLineParser as AssemblyRewriter
is signed and Mono.Options is not signed, resulting in compilation
failing when targeting net472.

russcam committed:
Add signing rewritten assemblies on Full Framework

This commit adds the ability to sign the rewritten assembly
when running on Full Framework. Signing is not currently
supported by Mono.Cecil on Net Core.

When a keyfile is supplied and merge is not specified (false),
the keyfile will be used to sign the rewritten assembly. When
merge is specified, the keyfile will be used to sign the merged
assembly.

Add net472 as a TargetFramework to allow signing functionality.

Move from Mono.Options to CommandLineParser as AssemblyRewriter
is signed and Mono.Options is not signed, resulting in compilation
failing when targeting net472.

russcam commented:
GitHub actions will need to run on windows-latest or windows-2019 and [have the net472 targeting pack installed](https://docs.microsoft.com/en-us/dotnet/framework/install/guide-for-developers) in order to compile net472 TFM

Mpdreamz commented:
I think we can support signing on .net core:
https://github.com/jbevain/cecil/pull/548

Mpdreamz commented:
See also: https://github.com/dsplaisted/strongnamer/pull/41/files

russcam committed:
Update Mono.Cecil to support signing on Net core

This commit updates Mono.Cecil to support signing on .NET Core

russcam committed:
Allow multiple instances of command line flags

russcam committed:
Rename explicit interface properties and methods

russcam committed:
Additional Rewrites

This commit adds additional rewrites for

1. explicitly implemented interface properties
2. explicitly implemented interface methods

Refactor the common name replacement. This can
probably be refactored further.

Verified rewriting with ildasm to il and searching for old names
Verified with peverify

// ignore some error codes that appear to be false positives
```
peverify <renamed>.dll /ignore=0x80131252,0x80131B2C /hresult /transparent
```

russcam commented:
Updated  PR to allow signing cross-platform on .NET Core
